iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -284,4 +333,40 @@ private static String getCatalogType(Configuration conf, String catalogName) {
       }
     }
   }
+
+  public static class TableCache {
+    private static final Cache<String, Table> tableCache = Caffeine.newBuilder()
+        .expireAfterAccess(12, TimeUnit.HOURS).build();
+    private static final Set<String> keys = new TreeSet<>();
+
+    public static void removeTable(Configuration conf) {
+      String queryId = conf.get(HiveConf.ConfVars.HIVEQUERYID.varname);
+      if (queryId != null && !queryId.isEmpty()) {
+        Set<String> queryKeys = keys.stream().filter(k -> k.startsWith(queryId)).collect(Collectors.toSet());
+        tableCache.invalidateAll(queryKeys);
+        keys.removeAll(queryKeys);
+      }
```

Comment:
shall we add an else branch here with some warn logging? It might be a problem here worth noting if the query id is not present in the config, because that will cause cache pollution over time. In the other methods, if the query id is not present we just reload the table, so that's not as big of a deal

Reply:
Correct, added logging.